### PR TITLE
added test for multipart/form-data request with text and file

### PR DESCRIPTION
**Pull request by theliuk**
I tried to reproduce [this bug](https://github.com/fastify/fastify-reply-from/issues/118) by means of a test. 
"Unfortunately" this new test passes and I wasn't able to reproduce the bug.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
I think you might want to add `fastify-multipart` directly... I think this plugin is incompatible with it.

**Comment by theliuk** [comment]
> I think you might want to add `fastify-multipart` directly... I think this plugin is incompatible with it.

Indeed: If the "proxying instance" (using `fastify-reply-from`) registers `fastify-multipart`, the "target instance" doesn't get multipart data.

**Comment by mcollina** [comment]
The problem is that we do not have:

1. the original data on the wire
2. a way to transform the data attached to `req` back into a multipart format

The most idiomatic solution to this problem at the user level is to leverage encapsulation.

```js
const app = Fastify()

app.register(async function (app) {
  app.register(Multipart)
  app.post(...) // route that uses multipart
})

app.register(async function (app) {
  app.register(From)
  app.post(...) // route that proxy correctly
})
``` 

I think the best path forward is to provide a meaningful error message when the plugin is registered and add some documentation in the README. You can use the `app.hasRequestDecorator` as a check, or maybe use `app.hasContentTypeParser` check. This is probably a stop-gap solution that at least provide help to the users.

We could also add a way to let the user provide the serialization method of the body.

**Comment by theliuk** [comment]
> The problem is that we do not have:
> 
> 1. the original data on the wire
> 2. a way to transform the data attached to `req` back into a multipart format
> 
> The most idiomatic solution to this problem at the user level is to leverage encapsulation.
> 
> ```js
> const app = Fastify()
> 
> app.register(async function (app) {
>   app.register(Multipart)
>   app.post(...) // route that uses multipart
> })
> 
> app.register(async function (app) {
>   app.register(From)
>   app.post(...) // route that proxy correctly
> })
> ```
> 
> I think the best path forward is to provide a meaningful error message when the plugin is registered and add some documentation in the README. You can use the `app.hasRequestDecorator` as a check, or maybe use `app.hasContentTypeParser` check. This is probably a stop-gap solution that at least provide help to the users.
> 
> We could also add a way to let the user provide the serialization method of the body.

Ok, I'll do it.
Shall I keep the test which documents the incompatibility with `fastify-multipart`?

**Comment by mcollina** [comment]
I think so. I'd also add it in the README if anybody stumbles on this.

File: README.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ HTTP2 to HTTP is supported too.
 npm i fastify-reply-from
 ```
 
+## Compatibility with fastify-multipart
+`fastify-reply-from` and [`fastify-multipart`](https://github.com/fastify/fastify-multipart) should not be registered as sibling plugins nor shold be registered in plugins which have a parent-child relationship.<br> The two plugins are incompatible, in the sense that the behavior of `fastify-reply-from` might not be the expected one when the above-mentioned conditions are not respected.<br>
+However, the two plugins may be used within the same fastify instance, at the condition that they belong to disjoint branches of the fastify plugins hierarchy tree.
```
Comment (mcollina) [inline README.md:17]: It would be good to explain why they are incompatible, i.e. the original multipart request is consumed when parsed.

**Review by mcollina — COMMENTED** [review]
Good work, just two nits

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**